File: jvm/core/src/main/java/se/idkollen/client/models/BankIdNoBackchannelAuthRequest.java
package se.idkollen.client.models;

import org.jspecify.annotations.Nullable;

/**
 * Request body for starting a BankID NO backchannel authentication session.
 */
public class BankIdNoBackchannelAuthRequest {
    public String ssn = "";
    public @Nullable String callbackUrl;
    public @Nullable String refId;

    public BankIdNoBackchannelAuthRequest setSsn(String v) {
        ssn = v;
        return this;
    }

    public BankIdNoBackchannelAuthRequest setCallbackUrl(@Nullable String v) {
        callbackUrl = v;
        return this;
    }

    public BankIdNoBackchannelAuthRequest setRefId(@Nullable String v) {
        refId = v;
        return this;
    }
}

File: php/src/Models/BankIdNo/BankIdNoBackchannelAuthRequest.php
<?php

declare(strict_types=1);

namespace Idkollen\Client\Models\BankIdNo;

readonly class BankIdNoBackchannelAuthRequest
{
    public function __construct(
        public string $ssn,
        public ?string $callbackUrl = null,
        public ?string $refId = null,
    ) {}
}
